src/lib/es6.d.ts

```diff
@@ -127,7 +127,7 @@ interface ObjectConstructor {
       * @param target The target object to copy to.
       * @param source The source object from which to copy properties.
       */
-    assign<T, U>(target: T, source: U): T & U;
+    assign<T, U>(target: T, source: U | null | undefined): T & U;

```

Comment:
BTW I don't think `Object.assign` is "meant" to _explicitely_ accept `undefined`. It reflects the fact the the parameter is optional and that should probably be modelled with `assign<T>(target: T): T`, so that you can call `assign(x)`; or maybe `assign<T, U>(target: T, source?: U)`.

Note that removing `null` and `undefined` from being explicit in the signature doesn't change the fact that `assign(x, null)` is still valid.

Of course the right thing to do here would be variadics (2.1) since the real signature is `assign<T, ...U>(target: T, ...source: ...U): T & ...U`.